#161966964 Setup react project

gwako94 opened the pull request:
#### What does this PR do?
Sets up empty react project

#### Description of Task to be completed?
Have an empty react project set up

#### How should this be manually tested?
```
$ git clone https://github.com/andela/ah-jumanji-frontend.git
$ cd ah-jumanji-frontend
$ git checkout ch-setup-project-161966964
$ npm install
$ npm start
```
#### Checklist:
- [x] My code follows the style guidelines of this project
- [x] At least 2 people have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] My PR has one commit.

gitau-kinuthia committed:
ch(set-up-project): Set up empty react application

- Set up webpack
- Set up Babel
- Set up eslint
- Set up hot realoading

[Finishes #161966964]

mashafrancis commented on README.md:
```diff
@@ -0,0 +1,21 @@
+## SET UP
```
You can add the project title "Authors Haven" on the Readme as well.

brayovsky commented on .eslintrc.json:
```diff
@@ -0,0 +1,89 @@
+    "jquery": true
+  },
+  "rules": {
+    "quotes": 0,
```
I think we should have consistent string quotes(single quotes) use. A `quotes: ["warn", "single", { "allowTemplateLiterals": true, "avoidEscape": true }]`.

gwako94 replied:
@brayovsky we are referencing the Andela engineering playbook style guides.
Kindly have a look at it here: https://gist.github.com/sa-adebayo/fe846cfb1c8c16660a7343aa94d21d65

gwako94 replied:
> I think we should have consistent string quotes(single quotes) use. A `quotes: ["warn", "single", { "allowTemplateLiterals": true, "avoidEscape": true }]`.

Based on the deliberation we've had among the team members, we've resolved to have both `single` and `double` quotes in that different implementations would require either of the two, e.g `strings` and `json data`.

brayovsky commented on .eslintrc.json:
```diff
@@ -0,0 +1,89 @@
+  },
+  "rules": {
+    "quotes": 0,
+    "new-cap": 0,
```
It's best practice to leave this on.

gwako94 replied:
> It's best practice to leave this on.

Done

brayovsky commented on .eslintrc.json:
```diff
@@ -0,0 +1,89 @@
+    "new-cap": 0,
+    "no-console": 1,
+    "no-debugger": 1,
+    "no-param-reassign": 0,
```
It's also best practice to have constructors begin with a capital letter

brayovsky commented on .eslintrc.json:
```diff
@@ -0,0 +1,89 @@
+    "no-unused-vars": 0,
+    "no-underscore-dangle": 0,
+    "no-alert": 0,
+    "no-lone-blocks": 0,
```
Line 37 - 42 also should be on as best practice

gwako94 replied:
> Line 37 - 42 also should be on as best practice

Done

gitau-kinuthia committed:
ch(set-up-project): Set up empty react application

- Set up webpack
- Set up Babel
- Set up eslint
- Set up hot reloading
- fix code climate errors

[Finishes #161966964]

GransonO reviewed (approved):
The PR works as expected.
The CI issues will be resolved once the testing has been set up
`LGTM`
